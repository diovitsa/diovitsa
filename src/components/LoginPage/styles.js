export const LoginPageStyles = {
  root: {
    width: '300px',
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  header: {
    font: '14px',
    fontWeight: 'bold',
  },
  wrongCreds: {
    font: '14px',
    fontWeight: 'bold',
    color: 'red',
    display: 'none'
  }

};
