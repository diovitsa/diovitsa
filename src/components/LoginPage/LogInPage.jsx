import React from 'react';
import withStyles from '@material-ui/core/styles/withStyles';
import { LoginPageStyles } from './styles';
import LoginForm from './LogInForm/LogInForm';
import { DataService } from '../../services/data/DataService';

class LogInPage extends React.Component {

  signIn = (email, password) => {
    return DataService.signIn(email, password)
      .then(() => {
        if (this.erorMessage) this.erorMessage.removeAttribute('style');
        this.props.history.push('/list');
      })
      .catch(() => {
        this.erorMessage = document.querySelector('#errorMessage');
        this.erorMessage.setAttribute('style', 'display: block');
      });
  };

  render() {
    const { root, header, wrongCreds } = this.props.classes;
    return (
      <div className={root}>
        <p className={header}> Welcome to my test app </p>
        <LoginForm onSignIn={this.signIn}/>
        <p id='errorMessage' className={wrongCreds}> Email or password is incorrect </p>
      </div>
    );
  }
}

export default withStyles(LoginPageStyles)(LogInPage);
