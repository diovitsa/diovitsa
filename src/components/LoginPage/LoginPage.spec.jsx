import React from 'react';
import { shallow } from 'enzyme';
import LoginPage from './LogInPage';
import { DataService } from '../../services/data/DataService';

describe('LogInForm', () => {
  let component;

  const history = {
    push: jest.fn()
  };

  function getComponent() {
    const testingComponent = shallow(<LoginPage history={history}/>).dive();
    return testingComponent.instance();
  }

  beforeEach(() => {
    component = getComponent();
  });

  it('should render properly', () => {
    expect(getComponent()).toMatchSnapshot();
  });


  describe('signIn', () => {

    beforeEach(() => {
      jest.spyOn(DataService, 'signIn').mockImplementation(() => Promise.resolve());
    });

    it('should call DataService.signIn', () => {
      component.signIn('email@gmail.com', 'password');
      expect(DataService.signIn).toHaveBeenCalledWith('email@gmail.com', 'password');
    });

    describe('on success', () => {
      it('should redirect user to UserList view', done => {
        component.signIn().then(() => {
          expect(component.props.history.push).toHaveBeenCalledWith('/list');
          done();
        });
      });
    });

  });
});
