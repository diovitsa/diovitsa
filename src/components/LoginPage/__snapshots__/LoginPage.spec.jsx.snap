// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogInForm should render properly 1`] = `
LogInPage {
  "context": Object {},
  "props": Object {
    "classes": Object {
      "header": "LogInPage-header-2",
      "root": "LogInPage-root-1",
      "wrongCreds": "LogInPage-wrongCreds-3",
    },
    "history": Object {
      "push": [MockFunction],
    },
  },
  "refs": Object {},
  "setState": [Function],
  "signIn": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <LogInPage
        classes={
          Object {
            "header": "LogInPage-header-2",
            "root": "LogInPage-root-1",
            "wrongCreds": "LogInPage-wrongCreds-3",
          }
        }
        history={
          Object {
            "push": [MockFunction],
          }
        }
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="LogInPage-root-1"
      >
        <p
          className="LogInPage-header-2"
        >
           Welcome to my test app 
        </p>
        <WithStyles(LogInForm)
          onSignIn={[Function]}
        />
        <p
          className="LogInPage-wrongCreds-3"
          id="errorMessage"
        >
           Email or password is incorrect 
        </p>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
  Symbol(enzyme.__setState__): [Function],
}
`;
